boba/parser.py

```diff
@@ -249,14 +249,14 @@ def _code_gen_recur(self, path, i, code, history):
 
             if chunk.variable != '':
                 # check if we have already encountered the placeholder variable
-                prev_idx = None
+                prev_option = None
                 for d in history.decisions:
                     if d.parameter == chunk.variable:
-                        prev_idx = d.idx
+                        prev_option = d.option
 
-                if prev_idx is not None:
+                if prev_option is not None:
```

Comment:
does `prev_option` has a fixed type (e.g. string)? We need to make sure a user-defined option will never be `None`. And if you use `==` with two options somewhere else, it might not work for types like floating point ... I used index before because it is an integer and we do not have to worry about these.